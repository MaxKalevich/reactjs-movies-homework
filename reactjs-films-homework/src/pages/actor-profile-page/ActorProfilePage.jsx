import ActorInfo from "../../components/actor-info/ActorInfo";
import MovieCard from "../../components/movie-card/MovieCard";

import { ActorProfilePageSideEffects } from "./actorProfilePageSideEffects";

import styles from "./actor-profile-page_style.module.scss";

const ActorProfilePage = () => {
  const { actorData, knownBy, actorImages, funcSetPage } =
    ActorProfilePageSideEffects();

  return (
    <div className={styles.contentContainer}>
      <div className={styles.detailsGroup}>
        {actorData !== undefined &&
          actorData.map((actor) => {
            return (
              <ActorInfo
                photo={actor.profile_path}
                name={actor.name}
                birthday={actor.birthday}
                placeOfBirthday={actor.place_of_birth}
                biography={actor.biography}
                photos={actor.photo}
                key={actor.id}
                actorImages={actorImages}
              />
            );
          })}
      </div>
      <div className={styles.knownBy}>
        <h2 className={styles.heading}>Known By</h2>
        <div className={styles.knownByContainer}>
          {knownBy !== undefined &&
            knownBy.map((movie) => {
              return (
                <MovieCard
                  movieTitle={movie.title}
                  movieImage={movie.poster_path}
                  estimation={movie.vote_average}
                  genre={movie.genre}
                  key={movie.title}
                  setPage={funcSetPage}
                />
              );
            })}
        </div>
      </div>
    </div>
  );
};

export default ActorProfilePage;
