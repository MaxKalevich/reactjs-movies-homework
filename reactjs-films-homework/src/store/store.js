import { combineReducers, configureStore } from "@reduxjs/toolkit";
import mainPageSlice from "./reducers/mainPageSlice";
import appSlice from "./reducers/appSlice";
import movieDetailsPageSlice from "./reducers/movieDetailsPageSlice";
import actorPageSlice from "./reducers/actorPageSlice";

const rootReducer = combineReducers({
  mainPageSlice,
  appSlice,
  movieDetailsPageSlice,
  actorPageSlice,
});

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(),
  devTools: process.env.NODE_ENV !== "production",
});
