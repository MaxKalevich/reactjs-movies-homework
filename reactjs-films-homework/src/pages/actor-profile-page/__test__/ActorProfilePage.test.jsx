import { render, screen } from "@testing-library/react";
import ActorProfilePage from "../ActorProfilePage";
import { Provider } from "react-redux";
import { store } from "../../../store/store";

describe("Checking ActorProfilePage component", () => {
  it("render ActorProfilePage component", async () => {
    const actorProfilePageMockData = [];
    const actorProfilePageMainActorMockData = [];
    render(
      <Provider store={store}>
        <ActorProfilePage
          actorProfilePageMockData={actorProfilePageMockData}
          actorProfilePageMainActorMockData={actorProfilePageMainActorMockData}
        />
      </Provider>
    );
    const linkElement = await screen.findByText(/Known By/i);
    expect(linkElement).toBeInTheDocument();
    const mainPageMockData = [];
    const { asFragment } = render(
      <Provider store={store}>
        <ActorProfilePage mainPageMockData={mainPageMockData} />
      </Provider>
    );
    expect(asFragment(<ActorProfilePage />)).toMatchSnapshot();
  });
});
