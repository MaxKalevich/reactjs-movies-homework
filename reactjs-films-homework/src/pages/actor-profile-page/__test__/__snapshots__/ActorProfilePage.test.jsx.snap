// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking ActorProfilePage component render ActorProfilePage component 1`] = `
<DocumentFragment>
  <img
    alt="spinner"
    class="spinner"
    src="/download_spinner.svg"
  />
</DocumentFragment>
`;
